shared-module/audiofreeverb/Freeverb.c

```diff
@@ -0,0 +1,358 @@
+// This file is part of the CircuitPython project: https://circuitpython.org
+//
+// SPDX-FileCopyrightText: Copyright (c) 2025 Mark Komus
+//
+// SPDX-License-Identifier: MIT
+//
+// Based on FreeVerb - https://github.com/sinshu/freeverb/tree/main
+// Fixed point ideas from - Paul Stoffregen in the Teensy audio library https://github.com/PaulStoffregen/Audio/blob/master/effect_freeverb.cpp
+//
+#include "shared-bindings/audiofreeverb/Freeverb.h"
+
+#include <stdint.h>
+#include "py/runtime.h"
+#include <math.h>
+
+void common_hal_audiofreeverb_freeverb_construct(audiofreeverb_freeverb_obj_t *self, mp_obj_t roomsize, mp_obj_t damp, mp_obj_t mix,
+    uint32_t buffer_size, uint8_t bits_per_sample,
+    bool samples_signed, uint8_t channel_count, uint32_t sample_rate) {
+
+    // Basic settings every effect and audio sample has
+    // These are the effects values, not the source sample(s)
+    self->base.bits_per_sample = bits_per_sample; // Most common is 16, but 8 is also supported in many places
+    self->base.samples_signed = samples_signed; // Are the samples we provide signed (common is true)
+    self->base.channel_count = channel_count; // Channels can be 1 for mono or 2 for stereo
+    self->base.sample_rate = sample_rate; // Sample rate for the effect, this generally needs to match all audio objects
+    self->base.single_buffer = false;
+    self->base.max_buffer_length = buffer_size;
+
+    // To smooth things out as CircuitPython is doing other tasks most audio objects have a buffer
+    // A double buffer is set up here so the audio output can use DMA on buffer 1 while we
+    // write to and create buffer 2.
+    // This buffer is what is passed to the audio component that plays the effect.
+    // Samples are set sequentially. For stereo audio they are passed L/R/L/R/...
+    self->buffer_len = buffer_size; // in bytes
+
+    self->buffer[0] = m_malloc(self->buffer_len);
```

Comment:
`m_malloc` will raise an exception on its own. If you want null back then use the `_maybe` version.

```suggestion
    self->buffer[0] = m_malloc_maybe(self->buffer_len);
```

Reply:
Fixed all the times this occurred.